palantirnet/the-vagrant
# Separate settings from the Vagrantfile

the-vagrant now uses a configuration yml, rather than copying the core Vagrantfile code into the project repository.

**Pros**

* We can fix bugs in the Vagrantfile and projects can easily update their copy of the-vagrant
* The configuration file is yml instead of variables in the Vagrantfile
* Phing is no longer a dependency

**Cons**

* Projects can't customize their whole Vagrantfile
  * e.g. no passing additional variables to Ansible, no customizing the disk mount, etc.
* Projects can't copy in the full provisioning anymore (but they can still add a custom playbook)
* Also this moves around the directory structure, so existing installs will break unless they reinstall

I think this change warrants a 3.x release. It's possible there's more cleanup to do here.

Other things that we have the opportunity to do here:
- [x] Update the installer wizard. It now creates a `.the-vagrant.yml`, but that's not required if you don't want to do any configuration. Should the wizard ask whether you want to customize at all?
- [x] Also we could rewrite the wizard in Ruby, which would remove phing as a dependency and remove the bash script wrapper
- [x] We have the opportunity to rename the configuration variables
- [x] We could only write configuration variables that have been customized to the yaml config
- [ ] ~Add some comments to the yaml config so that it's easier to customize after the fact~
- [x] Consider eliminating the option to entirely customize the provisioning -- in this case, you might as well not use the-vagrant...?

Test cases:
1. Install defaults
2. Install but customize settings
3. Install with custom playbook
4. Install and customize settings, then manually update
  * ... and break the yaml formatting
5. Install and customize the settings, setting the web root to a non-existent directory
  * ... note that the non-existent directory isn't saved, the existing directory is used instead
6. Install with custom playbook
  * ... then delete the custom playbook, and see if you can still vagrant up
